### PR TITLE
Implemented method ROCK4

### PR DESCRIPTION
**Pull request by deeepeshthakur**
@ChrisRackauckas @YingboMa Implemented ROCK4 from original Fortran Code. It's passing all the convergeence tests.
**Convergence Tests**
```
import DiffEqProblemLibrary.ODEProblemLibrary: prob_ode_linear
dts = 1 .//2 .^(8:-1:4)
testTol = 0.1
sim = test_convergence(dts,prob_ode_linear,ROCK4())
plot(sim)
```
image

```
import DiffEqProblemLibrary.ODEProblemLibrary: prob_ode_2Dlinear
dts = 1 .//2 .^(8:-1:4)
testTol = 0.1
sim = test_convergence(dts,prob_ode_2Dlinear,ROCK4())
plot(sim)
```
image

#2

### CODE REVIEW

File: test/ode/rkc_tests.jl [hunk 1]
```diff
@@ -29,3 +29,26 @@ end
     @test sim.𝒪est[:l∞] ≈ 2 atol=testTol
   end
 end
+
+@testset "Power Iteration of Runge-Kutta-Chebyshev Tests" begin
```
Comment (YingboMa) [inline test/ode/rkc_tests.jl]: You don't need to make a new test set here. Just append it in the above tests.

Comment (deeepeshthakur) [reply test/ode/rkc_tests.jl]: > You don't need to make a new test set here. Just append it in the above tests.

Updated.

File: test/ode/rkc_tests.jl [hunk 2]
```diff
@@ -28,4 +40,9 @@ end
     sim = test_convergence(dts,prob,ROCK2())
     @test sim.𝒪est[:l∞] ≈ 2 atol=testTol
   end
+
+  for prob in probArr
```
Comment (YingboMa) [inline test/ode/rkc_tests.jl]: You don't need to have two for loops here.

File: test/ode/rkc_tests.jl [hunk 3]
```diff
@@ -19,6 +19,18 @@ Random.seed!(123)
     eigest = integrator.eigen_est
     @test eigest ≈ eigm rtol=0.1eigm
   end
+
+  for i in 1:10, iip in [true, false]
```
Comment (YingboMa) [inline test/ode/rkc_tests.jl]: Just do `for i in 1:10, iip in [true, false], alg in [ROCK2(), ROCK4()]` and `integrator = init(prob, ROCK4())`

File: src/tableaus/rkc_tableaus.jl [hunk 4]
```diff
@@ -2277,3 +2277,2363 @@ function ROCK2ConstantCache(::Type{T}, ::Type{T2}, zprev) where {T, T2}
...
+         (0.001825777275563891, 0.6395518371737207)
+         (0.001834853595977716, 0.6468267831733791)
+         (0.001843526627209035, 0.6537971564070679)
+         (0.001851822970482422, 0.6604817024077319)
+         (0.001859767092366030, 0.6668977957794100)
+         (0.001867381581918830, 0.6730615950017611)
+         (0.001874687391723729, 0.6789881919521959)
+         (0.001881704069826390, 0.6846917603173716)
+         (0.001888449989971118, 0.6901857076183885)
+         (0.001894942588218185, 0.69548283633
```
Comment (ChrisRackauckas) [inline src/tableaus/rkc_tableaus.jl]: this is quite the constant

**Comment by ChrisRackauckas** [comment]
It looks good to me. I closed/reopened to re-run CI.

**Pull request merged**